Code/Framework/AzFramework/Platform/Windows/AzFramework/Process/ProcessWatcher_Win.cpp

```diff
@@ -355,4 +356,48 @@ namespace AzFramework
             ::TerminateProcess(m_pWatcherData->processInformation.hProcess, exitCode);
         }
     }
+
+        namespace ProcessLauncher
+    {
+        AZStd::string ProcessLaunchInfo::GetCommandLineParametersAsString() const
+        {
+            struct CommandLineParametersVisitor
+            {
+                AZStd::string operator()(const AZStd::string& commandLine) const
+                {
+                    return commandLine;
+                }
+
+                AZStd::string operator()(const AZStd::vector<AZStd::string>& commandLineArray) const
+                {
+                    AZStd::string commandLineResult;
+
+                    // When re-constructing a command line from an argument list (on windows), if an argument
+                    // is double-quoted, then the double-quotes must be escaped properly otherwise
+                    // it will be absorbed by the native argument parser and possibly evaluated as
+                    // multiple values for arguments
+                    AZStd::string_view escapedDoubleQuote = R"("\")";
+
+                    AZStd::vector<AZStd::string> preprocessed_command_array;
+                    for (const auto& commandArg : commandLineArray)
+                    {
+                        AZStd::string replacedArg = commandArg;
+                        auto replacePos = replacedArg.find_first_of('"');
+                        while (replacePos != replacedArg.npos)
+                        {
+                            replacedArg.replace(replacePos, 1, escapedDoubleQuote.data(), escapedDoubleQuote.size());
+                            replacePos += escapedDoubleQuote.size();
+                            replacePos = replacedArg.find('"', replacePos);
+                        }
+                        preprocessed_command_array.emplace_back(replacedArg);
```

Comment:
The `AZ::StringFunc::Replace` should be used here.

That would also get rid of the `preprocessed_command_array` variable which is doesn't meet the variable naming style requirements.

```suggestion
                    for (AZStd::string& commandArg : commandLineArray)
                    {
                        AZ::StringFunc::Replace(commandArg, R"(")", R"(\")");
                    }
```